mava/core_jax_test.py

```diff
@@ -0,0 +1,108 @@
+# python3
+# Copyright 2021 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""Tests for core Mava interfaces for Jax systems."""
+
+from types import SimpleNamespace
+from typing import Any, List
+
+import pytest
+
+from mava.core_jax import BaseSystem
+
+# Dummy component
+COMPONENT = None
+
+
+def test_exception_for_incomplete_child_class() -> None:
```

Comment:
If the class is abstract this is expected behavior so what is the purpose of this test? To make sure that over time, we don't remove the abstract annotations?

Reply:
Yes, I think that is the purpose of most tests, to ensure the behaviour of code is maintained over time. Of course this is quite small and it is unlikely that anyone remove some of the decorators, but writing the tests are cheap so I think it is still worth having this extra layer of checks.